test/writer_test.rb

```diff
@@ -0,0 +1,38 @@
+require 'helper'
+require 'ddtrace/transport'
+require 'ddtrace/writer'
+require 'json'
+
+module Datadog
+  class WriterTest < Minitest::Test
+    def test_sampling_feedback_loop
+      sampler = Minitest::Mock.new
+      writer = Writer.new(priority_sampler: sampler)
+      response_body = { 'service_a' => 0.1, 'service_b' => 0.5 }.to_json
```

Comment:
Real-world responses are more like:
`{  'service:,env:' => 0.1, 'service:a,env:test' => 0.3, 'service:b,env:test' => 0.15 }`
while, technically, it's perfectly fine to test it with the entry you used, I also think it's good that the test data, be it mocked, actually reflect real use cases, so it would be nice to have at least one test that has real data (or at least, data that is technically likely to happen). On the agent you get hints on this here https://github.com/DataDog/datadog-trace-agent/blob/master/sampler/catalog_test.go#L66 or there https://github.com/DataDog/datadog-trace-agent/blob/master/agent/info_test.go#L107

Reply:
👍  I agree 100%